include/aspect/compat.h

```diff
@@ -23,6 +23,26 @@
 
 #include <aspect/global.h>
 
+#if DEAL_II_VERSION_GTE(8,4,0)
+#else
+template <int dim>
+dealii::Tensor<1,dim> cross_product(const dealii::Tensor<1,dim> &a, const dealii::Tensor<1,dim> &b)
```

Comment:
Ah, I see -- you need it closer to the ASPECT code to make sure it finds this place first? Fair enough. Go with it then!